### ISSUE TITLE
[24] Monitor failures in tests of Y-Builds

### ISSUE THREAD

**Issue by stephan-herrmann** [eclipse-jdt/eclipse.jdt.core#3228]
In particular this job should be closely monitored for [24] work:
* https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/

Unfortunately there's a HUGE number of failures outside JDT (mostly against pde.api.tools), but when you scroll down to the per-package statistics, it's possible to check just everything in org.eclipse.jdt.core*.

Status as of https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/7/testReport/
* zero failures in org.eclipse.jdt.{apt,compiler,core} tests

**Comment by stephan-herrmann** [comment]
> HUGE number of failures outside JDT (mostly against pde.api.tools),

Is this something that happened before for previous Java versions? Perhaps api tools simply can't read class files of Java 24?
* Should PDE join the BETA_JAVA24 train for this purpose? 
* Or is there a simple way to exclude those tests from the build?

**Comment by stephan-herrmann** [comment]
For monitoring only jdt.core it is easier to just look at this job:
* https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/job/BETA_JAVA24/

**Comment by stephan-herrmann** [comment]
> > HUGE number of failures outside JDT (mostly against pde.api.tools),
> 
> Is this something that happened before for previous Java versions? Perhaps api tools simply can't read class files of Java 24?

Much of this was resolved, I assume be changes in jdt.lauching.

**Comment by stephan-herrmann** [comment]
We have a drop in number of tests, see https://ci.eclipse.org/releng/job/YPBuilds/job/ep434Y-unit-cen64-gtk3-java24/14/testReport/org.eclipse.jdt.core.tests.compiler.parser/TestAll/ which, however can be explained by the removal of SelectionParserTest12.java via #3264

**Comment by HannesWell** [comment]
> While JDT/Core's BETA_JAVA24 build is green, [Y-Build tests on Java 24](https://ci.eclipse.org/releng/job/YPBuilds/job/ep435Y-unit-linux-x86_64-java24/) abort since 2024-12-31.

Looks like your change in https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3517 really fixed the test result collection as you described in https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2625#issuecomment-2571698112 🎉

**Comment by stephan-herrmann** [comment]
Indeed we have fresh test results! :)

Meanwhile the following tests in jdt.core fail (while locally and during JDT-builds they pass):
* org.eclipse.jdt.core.tests.model.Java9ElementTests.test515342b
* org.eclipse.jdt.core.tests.model.Java9ElementTests.test530024_001
* org.eclipse.jdt.core.tests.model.Java9ElementTests.testSystemLibAsJMod
* org.eclipse.jdt.core.tests.model.ModuleBuilderTests.testSystemLibAsJMod
* org.eclipse.jdt.core.tests.model.ModuleBuilderTests.testSystemLibAsJMod_2

**Comment by stephan-herrmann** [comment]
> > Indeed we have fresh test results! :)
> > Meanwhile the following tests in jdt.core fail (while locally and during JDT-builds they pass):
> 
> All of them appear to be related to modules.

Bingo: I can locally reproduce using temurin 24+30 JDK (but not using openjdk 24+29)

**Comment by stephan-herrmann** [comment]
@jarthana Java9ElementTests.test515342b (and possibly others) explicitly tries to read *jdk*/jmods/java.base.jmod, but the temurin JDK used during production builds does not have any jmods.

https://bugs.eclipse.org/bugs/show_bug.cgi?id=515342 doesn't mention jmods, so do you by any chance remember (after more than 7 years ...) why the test was written like this?

**Comment by stephan-herrmann** [comment]
@MohananRahul  @HannesWell could we switch to running tests on openjdk ea 24, rather than the temurin flavor?

**Comment by HannesWell** [comment]
> @MohananRahul  @HannesWell could we switch to running tests on openjdk ea 24, rather than the temurin flavor?

Yes absolutely. Anything that can be downloaded via a URL, preferably in tar.gz format can be used. 

Can you provide the URL?

**Comment by jarthana** [comment]
> @jarthana Java9ElementTests.test515342b (and possibly others) explicitly tries to read _jdk_/jmods/java.base.jmod, but the temurin JDK used during production builds does not have any jmods.
> 
> https://bugs.eclipse.org/bugs/show_bug.cgi?id=515342 doesn't mention jmods, so do you by any chance remember (after more than 7 years ...) why the test was written like this?

This may have been some leftover from the effort of [bug 490103](https://bugs.eclipse.org/bugs/show_bug.cgi?id=490103) that tried to get rid of the bogus jrt-fs.jar that represents the JRE container. But in the end, we decided it depended too much on the implementation details of the JDK.

**Comment by stephan-herrmann** [comment]
> > @jarthana Java9ElementTests.test515342b (and possibly others) explicitly tries to read _jdk_/jmods/java.base.jmod, but the temurin JDK used during production builds does not have any jmods.
> > https://bugs.eclipse.org/bugs/show_bug.cgi?id=515342 doesn't mention jmods, so do you by any chance remember (after more than 7 years ...) why the test was written like this?
> 
> This may have been some leftover from the effort of [bug 490103](https://bugs.eclipse.org/bugs/show_bug.cgi?id=490103) that tried to get rid of the bogus jrt-fs.jar that represents the JRE container. But in the end, we decided it depended too much on the implementation details of the JDK.

Trying to refresh memory of old discussions I came across https://bugs.eclipse.org/bugs/show_bug.cgi?id=490103#c65

Do you think we should remove the implementation for jmod support or only disable / remove those tests?

**Comment by stephan-herrmann** [comment]
https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/2738 did the trick: As of https://ci.eclipse.org/releng/job/YPBuilds/job/ep435Y-unit-linux-x86_64-java24/26/testReport/ jdt.core is clean again! :)

**Comment by stephan-herrmann** [comment]
Listing sibling issues for convenience:
* JDT/UI: https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/1763 (currently clean)
* JDT/Debug: https://github.com/eclipse-jdt/eclipse.jdt.debug/issues/553 (9 failures)

**Comment by stephan-herrmann** [comment]
Meanwhile also the build result page is built, found a recent one here
*  overview: https://download.eclipse.org/eclipse/downloads/drops4/Y20250110-1000/
* test results: https://download.eclipse.org/eclipse/downloads/drops4/Y20250110-1000/testResults.php

Does anyone know if those are linked from some index page?

**Comment by jarthana** [comment]
> Meanwhile also the build result page is built, found a recent one here
> 
>     * overview: https://download.eclipse.org/eclipse/downloads/drops4/Y20250110-1000/
> 
>     * test results: https://download.eclipse.org/eclipse/downloads/drops4/Y20250110-1000/testResults.php
> 
> 
> Does anyone know if those are linked from some index page?

AFAIK, we never published these. May be we can when there's demand and if we know anyone (other than the committers) is going to be downloading them.

Meanwhile, I have two points:
1. I thought I will keep the jmod related code for reasons stated [here](https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3587#issuecomment-2604139763) and adjust the tests for Temurin. But when I downloaded the JDK 21 for windows, I still see the jmods in the JDK. Wonder if it's OS specific. Not a big deal for me, but I am unable to test the test changes.
2. Now, I am slowly recollecting some of the thought process behind those tests. I believe I wanted to avoid all modules being loaded by the model tests and therefore cause delay. Keeping only the jmods in the classpath should help tests complete quicker.

**Comment by stephan-herrmann** [comment]
> 1. But when I downloaded the JDK 21 for windows, I still see the jmods in the JDK.

I took another sample, and downloaded from adoptium: OpenJDK24U-jdk_x64_windows_hotspot_24_33-ea.zip

```
$ unzip -l OpenJDK24U-jdk_x64_windows_hotspot_24_33-ea.zip |grep jmod
    23656  2025-01-23 23:22   jdk-24+33/bin/jmod.exe
```

Not jmod files here.

**Comment by jarthana** [comment]
> ```
> $ unzip -l OpenJDK24U-jdk_x64_windows_hotspot_24_33-ea.zip |grep jmod
>     23656  2025-01-23 23:22   jdk-24+33/bin/jmod.exe
> ```
> 
> Not jmod files here.

I can confirm that here with jdk 24. Thanks. Now I am able to test and also noticed a silly error with my code and test was failing. Fixing it with https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3632

**Comment by stephan-herrmann** [comment]
As of https://ci.eclipse.org/releng/job/YPBuilds/job/ep435Y-unit-linux-x86_64-java24/68/testReport/ we have 40 failures in api tools. Let's hope/assume that these are related only to the recent switch to 4.36 + R4_35_maintenance and not indicative of a real code problem.

**Comment by stephan-herrmann** [comment]
Today's build failed at [org.eclipse.jdt.core.tests.model.ClasspathTests.testDotDotLibraryEntry8](https://ci.eclipse.org/releng/job/YPBuilds/job/ep435Y-unit-linux-x86_64-java24/69/testReport/org.eclipse.jdt.core.tests.model/ClasspathTests/testDotDotLibraryEntry8/) -- I believe this test is unstable also on master, right?

**Comment by stephan-herrmann** [comment]
With [final tests passing](https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2899#issuecomment-2734404255) we're done here for 24.
